### DISCUSSION TITLE
Relay to switch external lights simultanously to the internal light of the motor

### DISCUSSION THREAD

**Discussion by asdt1803** [Gifford47/HCPBridgeMqtt#44] [Ideas]
name | about | title | labels | assignees
-- | -- | -- | -- | --
Feature request | Suggest an idea for this project | Switch exernal lights  | enhancement |  

<b>Is your feature request related to a problem?</b> The original implementation https://github.com/hkiam/HCPBridge featured an relay to switch external lights. Unfortunately this feature is currently missing in your implementation.

<b>Describe the solution you'd like</b> Use a digital output to switch a relay simultaneously to the internal light of the motor to switch additional lights in the garage or driveway. Optimally use an LDR on an analogue input to switch on the external lights only when the ambient light is low. Input and output port should be made configurable in cofiguration.h.

<b>Describe alternatives you've considered</b> Use the MQTT output and a script to switch an Tasmota plug. This would require additional external hardware.

<b>Additional context</b> I don't use HomeAssistant and would prefer to have the relay integrated into the housing of my HCPbridge.

**Comment by 14yannick** [comment]
Hi, yes it really looks like this was lost in the multiple revisions. As alternative you can try the [esphome](https://github.com/mapero/esphome-hcpbridge/tree/main) port. This will give you the possibility to trigger an action on light [switch](https://esphome.io/components/switch/index.html#switch-on-turn-on-off-trigger). so you set the gpio output there

**Comment by fjaraujo** [comment]
Hi, in my case I need that feature to actuate over an [electric lock](https://www.viro.it/en-gb/hardware/locks-electric-strikes/v09) installed at the bottom of the door. Here the timing is critical, the relay must be activated before the door starts to move, otherwise will be blocked. The approach is to configure a pre-warning action in the menu 30 of the motor, saying that launch the signal 1s before the movement. I have managed to debug the messages within the code, but nothing seems to be sent by the motor... so I don't see how to catch the message and actuate over a digital signal in the HCPBridge board.

**Comment by scastoldiprivate** [comment]
Hello,
I'm also interested in a version with a relay to switch external light.
Thanks to @vladimirzrnic for designing a PCB.
Is anyone aware of a PCB with relay that could do the job ?
Thanks

**Comment by 14yannick** [comment]
Some news about this topic I think I was able to interpret the Status of the option relay. I implemented it [here](https://github.com/14yannick/esphome-hcpbridge) as a binary_sensor who show it's state (see example yaml). The option relay can be configured on the motor to get activated as pre-warning for example to triggered your door unlock @fjaraujo . You cans also configure it to get activated with the light with the same or a different off timer. The binary_sensor( option relay or light) can then trigger a [Relay](https://shop.m5stack.com/products/mini-3a-relay-unit).

**Reply by fjaraujo** [reply]
Great work @14yannick!  I installed this component ant now I can unlock the door using the new sensor. Thanks!
